pkg/syncer/syncertest/fake/client.go

```diff
@@ -356,15 +366,27 @@ func (c *Client) Delete(_ context.Context, obj client.Object, opts ...client.Del
 	}
 	obj.GetObjectKind().SetGroupVersionKind(gvk)
 
-	id := core.IDOf(obj)
+	id := c.idFromObject(obj)
 
 	cachedObj, found := c.Objects[id]
 	if !found {
 		return newNotFound(id)
 	}
 
-	klog.V(5).Infof("Deleting %T %s (ResourceVersion: %q): %s",
-		cachedObj, client.ObjectKeyFromObject(cachedObj), cachedObj.GetResourceVersion(), log.AsJSON(cachedObj))
+	if obj.GetUID() != "" && obj.GetUID() != cachedObj.GetUID() {
+		return newConflictingUID(id, obj.GetResourceVersion(), cachedObj.GetResourceVersion())
+	}
+	// Copy cached UID to updated object
+	obj.SetUID(cachedObj.GetUID())
```

Comment:
Why do we need to set the obj UID here?

Reply:
The fake client is simulating the apiserver, and the apiserver sets the UID on newly created objects. That way you can confirm that it's the same instance of the object you expected. If it's changed, that means the object was deleted and re-created.

So for creates we initialize it, and for updates we make sure the UID stays the same as what was previously created (from the in-memory cache, in this case).

Reply:
Oh, and when you Create, Update, or Delete an object with the normal client, it copies the updated values back to the input object, so you can read them, since there's no return object. That's what it's doing here.

Reply:
It looks like Delete is different from Create and Update, the typed client does not update the input object https://github.com/GoogleContainerTools/kpt-config-sync/blob/main/vendor/sigs.k8s.io/controller-runtime/pkg/client/typed_client.go#L74.  And I did not find any test in this PR using this UID after Delete.